setup.py

```diff
@@ -0,0 +1,83 @@
+# Copyright 2017 Mycroft AI Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+from setuptools import setup, find_packages
+import os.path
+
+BASEDIR = os.path.abspath(os.path.dirname(__file__))
+
+
+def get_version():
+    """ Find the version of mycroft-core"""
+    version = None
+    version_file = os.path.join(BASEDIR, 'mycroft', 'version', '__init__.py')
+    try:
+        major, minor, build = (None, None, None)
+        with open(version_file) as f:
+            for line in f:
+                if 'CORE_VERSION_MAJOR' in line:
+                    major = line.split('=')[1].strip()
+                elif 'CORE_VERSION_MINOR' in line:
+                    minor = line.split('=')[1].strip()
+                elif 'CORE_VERSION_BUILD' in line:
+                    build = line.split('=')[1].strip()
+
+                if ((major and minor and build) or
+                        '# END_VERSION_BLOCK' in line):
+                    break
+        version = '.'.join([major, minor, build])
+    except Exception as e:
+        print('Couldn\'t determine version ({})'.format(repr(e)))
+
+    return version
+
+
+def required(requirements_file):
+    with open(os.path.join(BASEDIR, requirements_file), 'r') as f:
+        requirements = f.read().splitlines()
+        return [pkg for pkg in requirements if not pkg.startswith("--")]
```

Comment:
Will this include empty lines and should the check be `if not pkg.startswith("#")`?